queries.sql

```diff
@@ -82,3 +82,55 @@ WHERE Escape_attempts = 0 AND owners.id = 5;
 SELECT animals.name,COUNT(owner_id),full_name FROM animals
 JOIN owners ON owners.id = animals.owner_id;
 
+SELECT animals.name,visits.date FROM animals
+JOIN visits ON animals.id = visits.animals_id
+JOIN vets ON vets.id = visits.vet_id
+WHERE vets.id = 1
+GROUP BY animals.name,visits.date
+ORDER BY date DESC LIMIT 1;
+
+SELECT COUNT(DISTINCT visits.animals_id) FROM visits
+JOIN vets ON vets.id visits.vet_id
+WHERE vets.id = 3;
+
+SELECT vets.name, species.name FROM vets
+JOIN specialization ON specialization.vets_id = vets.id
+JOIN species ON species.id = specialization.specie_id;
+
+SELECT animals.name, visits.date FROM visits
+    JOIN vets ON vets.id = visits.vet_id
+    JOIN animals ON animals.id = visits.animals_id
+    WHERE vets.name = 'Stephanie Mendez'
+    AND visits.date BETWEEN (to_date('04/01/2020','mm/dd/yyyy')) AND (to_date('08/30/2000','mm/dd/yyyy'));
+
+SELECT animals.name, COUNT(visits.animals_id) count_visits FROM visits
+JOIN vets ON vets.id = visits.vet_id
+JOIN animals ON animals.id = visits.animals_id
+GROUP BY animals.name,visits.animals_id
+ORDER BY count_visits DESC LIMIT 1;
+
+SELECT animals.name,visits.date AS first_visit FROM visits
+JOIN vets ON vets.id = visits.vet_id
+JOIN animals ON animals.id = visits.animals_id
+WHERE vets.id = 2
+GROUP BY animals.name,visits.date
+ORDER BY first_visit LIMIT 1;
+
+SELECT * FROM visits
+JOIN vets ON vets.id = visits.vet_id
+JOIN animals ON animals.id = visits.animals_id
+ORDER BY visits.date DESC LIMIT 1;
+
+SELECT COUNT(visits.animals_id) FROM visits
+JOIN vets ON vets.id = visits.vet_id
+JOIN animals ON animals.id = visits.animals_id
+JOIN specialization ON specialization.vets_id = vets.id
+WHERE specialization.specie_id <> animals.species_id;
+
+SELECT species.name,COUNT(visits.animals_id) AS count_species FROM visits
+JOIN vets ON vets.id = visits.vet_id
+JOIN animals ON animals.id = visits.animals_id
+JOIN species ON species.id = animals.species_id
+WHERE vets.id = 2
+GROUP BY species.name
+ORDER BY count_species DESC LIMIT 1;
```

Comment:
- [ ] Make sure the changes you are going to implement , doesn't affect your solution

Reply:
Sure